### ISSUE TITLE
[1.16.5] Rice crop crash

### ISSUE THREAD

**Issue by Snownee** [vectorwing/FarmersDelight#194]
**Steps to reproduce:**
I was deleting a chunk using World Stripper mod, and that chunk has rice crop

**Logs**
[crash-2021-02-18_11.46.56-server.txt](https://github.com/vectorwing/FarmersDelight/files/6000110/crash-2021-02-18_11.46.56-server.txt)

**Comment by vectorwing** [comment]
After looking at the offending line in your log, it's right on updatePostPlacement. I check if the block isn't air to do a fluid update, but immediately disregard it to add a SUPPORTING state under it if the block above isn't RiceUpperBlock.

This might be hard to test, but would it work if I moved this state change into the isAir check?
https://github.com/vectorwing/FarmersDelight/blob/447d3eb5bc6edad1869a509d024af9f252e10e3a/src/main/java/vectorwing/farmersdelight/blocks/RiceCropBlock.java#L124

**Comment by vectorwing** [comment]
I ran a local test and managed to replicate the exact crash you had. World Stripper seems to leave waterlogged, or watery, plants intact when stripping. I assume you tried punching rice and it crashed, right?

With the change I described above, punching the floating rice no longer crashes. I'll commit it soon; here's the screenshot of how the mod treats water plants. 😆

![image](https://user-images.githubusercontent.com/5566692/109405810-18c8ee80-7953-11eb-916f-438b0e7c43b8.png)
